--- a/board/samsung/smdk5422/smc.c	2015-11-28 21:03:06.257766485 -0800
+++ b/board/samsung/smdk5422/smc.c	2015-11-28 21:02:33.636342985 -0800
@@ -22,7 +22,7 @@
 	register u32 reg3 __asm__("r3") = arg3;
 
 	__asm__ volatile (
-#if __GNUC__ >= 4 && __GNUC_MINOR__ >= 6
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)
 		".arch_extension sec\n"
 #endif
 		"smc	0\n"
@@ -39,7 +39,7 @@
 	register u32 reg1 __asm__("r1") = 0;
 
 	__asm__ volatile (
-#if __GNUC__ >= 4 && __GNUC_MINOR__ >= 6
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)
 		".arch_extension sec\n"
 #endif
 		"smc	0\n"
